main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -374,7 +374,7 @@ object Ast {
   /**
     * Represents a label in the labelled graph.
     */
-  case class Label(pred: Name.Pred, arity: Int)
+  case class Label(pred: Name.Pred, terms: List[Type])
```

Comment:
Two things:

- Would it make sense to also include whether is a relation or a lattice?
- Would it perhaps be better to store a single type instead of a list of terms? (Or maybe its better like it is because then you can deal with arity quickly?)

Reply:
> Would it make sense to also include whether is a relation or a lattice?

Good point, hadn't thought of that.
> Would it perhaps be better to store a single type instead of a list of terms?

The representation of predicates terms are a bit weird since they are sometimes tuples, sometimes a single type and sometimes unit. So I thought it would be nice. We are probably need to change them at some point later I think because of this https://github.com/flix/flix/issues/2353. I think the term types are ambiguous since you can't differentiate a single term of tuple type and a predicate with multiple terms